hytra/core/hypothesesgraph.py

```diff
@@ -787,6 +787,42 @@ def pruneGraphToSolution(self, distanceToSolution=0):
                     prunedGraph._graph.add_node(dest,**self._graph.node[dest])
                     prunedGraph._graph.add_edge(src,dest,**self._graph.edge[src][dest])
 
+        # in case a node is NOT an appearance and
+        # has all the incoming edges with value 0, we remove all these incoming edges
+        #
+        # in case a node is NOT a disappearance and
+        # has all the outgoing edges with value 0, we remove all these outgoing edges
+        for n in self.nodeIterator():
+            if 'appearance' in self._graph.node[n].keys() and self._graph.node[n]['appearance']:
+                maxNumObjects = len(self._graph.node[n]['appearanceFeatures'])-1
+                self._graph.node[n]['appearanceFeatures'] = listify([0.0] + [1.0] * maxNumObjects)
```

Comment:
Actually setting the `appearanceFeatures` to 1 could be an error, because at the image boundary this value gets lower, and we might want to learn with the real value and not 1.